docs/standard/serialization/system-text-json/snippets/how-to/csharp/Utf8ReaderPartialRead.cs

```diff
@@ -7,74 +7,85 @@ public class Utf8ReaderPartialRead
 
             // Search for "Summary" property name
-            while (reader.TokenType != JsonTokenType.PropertyName || !reader.ValueTextEquals("Summary"))
+            while (reader.Read() || UpdateBuffer(stream, ref buffer, ref read, ref reader))
             {
-                if (!reader.Read())
+                if (reader.TokenType is JsonTokenType.PropertyName && reader.ValueTextEquals(searchText))
+                {
+                    foundTerm = true;
+                }
+                else if (foundTerm && reader.TokenType is JsonTokenType.String)
                 {
-                    // Not enough of the JSON is in the buffer to complete a read.
-                    GetMoreBytesFromStream(stream, ref buffer, ref reader);
+                    Console.WriteLine($"Found {Encoding.UTF8.GetString(searchText)} value: {reader.GetString()}");
+                    break;
                 }
             }
 
-            // Found the "Summary" property name.
-            Console.WriteLine($"String in buffer is: {Encoding.UTF8.GetString(buffer)}");
-            while (!reader.Read())
-            {
-                // Not enough of the JSON is in the buffer to complete a read.
-                GetMoreBytesFromStream(stream, ref buffer, ref reader);
-            }
-            // Display value of Summary property, that is, "Hot".
-            Console.WriteLine($"Got property value: {reader.GetString()}");
-        }
+            Console.WriteLine($"String in buffer is: {Encoding.UTF8.GetString(buffer[..read])}");
 
-        private static void GetMoreBytesFromStream(
-            MemoryStream stream, ref byte[] buffer, ref Utf8JsonReader reader)
-        {
-            int bytesRead;
-            if (reader.BytesConsumed < buffer.Length)
+            static bool UpdateBuffer(MemoryStream stream, ref Span<byte> buffer, ref int read, ref Utf8JsonReader reader)
             {
-                ReadOnlySpan<byte> leftover = buffer.AsSpan((int)reader.BytesConsumed);
+                var bytesConsumed = (int)reader.BytesConsumed;
+                var bufferStart = 0;
 
-                if (leftover.Length == buffer.Length)
+                if (bytesConsumed < buffer.Length)
                 {
-                    Array.Resize(ref buffer, buffer.Length * 2);
-                    Console.WriteLine($"Increased buffer size to {buffer.Length}");
+                    var leftOver = buffer[bytesConsumed..];
+                    leftOver.CopyTo(buffer);
+                    bufferStart = leftOver.Length;
                 }
 
-                leftover.CopyTo(buffer);
-                bytesRead = stream.Read(buffer.AsSpan(leftover.Length));
+                read = stream.Read(buffer[bufferStart..]);
+                var isFinal = read < bytesConsumed;
+                Console.WriteLine($"Updating buffer ... leftOver bytes: {bufferStart}; bytes read: {read}; isFinalBlock: {isFinal}");
+
+                if (read is 0)
```

Comment:
Nit: I tend to reserve `is` checks for polymorphic pattern matching (i.e. I would use it if the type of `read` were `object`, but would just write `== 0` since the type is `int`).

Reply:
The `is` check is used for many types of operations, not just polymorphic pattern matching. In many cases, like when checking for `null` or lack of `null`, `is` can be a much more direct check than `==`.  I agree for types like `int` it's a bit more a matter of style but there is nothing wrong with this.

Reply:
Agreed, it's just my own preference to avoid it when comparing two integers.